StripeConnect/StripeConnect/Source/Internal/Webview/ConnectComponentWebViewController.swift

```diff
@@ -123,19 +140,41 @@ class ConnectComponentWebViewController: ConnectWebViewController {
         }
     }
 
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        analyticsClient.logComponentViewed(viewedAt: .now)
+    }
+
     // MARK: - ConnectWebViewController
 
+    override func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+        super.webView(webView, didFinish: navigation)
+
+        analyticsClient.logComponentWebPageLoaded(loadEnd: .now)
+    }
+
     override func webView(_ webView: WKWebView, didFail navigation: WKNavigation!, withError error: any Error) {
         super.webView(webView, didFail: navigation, withError: error)
+
         didFailLoad(error: error)
-        // TODO: MXMOBILE-2491 log error
+        analyticsClient.log(event: PageLoadErrorEvent(metadata: .init(
+            error: error,
+            url: webView.url
+        )))
     }
 
     override func webView(_ webView: WKWebView, decidePolicyFor navigationResponse: WKNavigationResponse) async -> WKNavigationResponsePolicy {
+        // If the component web page fails to load with an HTTP error, send a
+        // load failure to event
         if let response = navigationResponse.response as? HTTPURLResponse,
+           response.url?.absoluteStringRemovingParams == StripeConnectConstants.connectJSBaseURL.absoluteString,
```

Comment:
Added this check to ensure we only call `didFailLoad` if the failure happens on the root component page. This method is called for any HTTP request originating from the page, including loading resources like image assets. If an image asset fails to load, we don't want to send a failure callback.